### ISSUE TITLE
Strikeout deprecated headings in sidemenu for docusaurus

### ISSUE THREAD

**Issue by DominicGBauer** [typedoc2md/typedoc-plugin-markdown#747]
### What package is the feature request related to?

docusaurus-plugin-typedoc

### Background

When using docusaurus plugin with sidemenu, the sidemenu title is not struck out when deprecated.
<img width="839" alt="image" src="https://github.com/user-attachments/assets/63e84e39-ead5-4a47-ac39-24d5ef689220" />

### Proposed solution

It would really be nice if the sidemenu item is struck out when deprecated. Example from dart docs
<img width="212" alt="image" src="https://github.com/user-attachments/assets/aa35c831-0008-4f00-bc39-5bb0aaee278e" />

**Comment by tgreyuk** [comment]
Yes can do this. Will need to expose an additional flag to navigation via typedoc-plugin-markdown too. Will add to list for next release.

**Comment by DominicGBauer** [comment]
Epic thanks @tgreyuk!

**Comment by tgreyuk** [comment]
This functionality is available in:

`docusaurus-plugin-typedoc@1.2.0`
`typedoc-plugin-markdown@4.4.0`

**Comment by DominicGBauer** [comment]
Hi @tgreyuk thanks a lot for this! Not to sound ungrateful but is it possible to make the strikeout more apparent and have the line not miss any space please?

<img width="265" alt="image" src="https://github.com/user-attachments/assets/3327d96a-e6a6-41ed-aa9d-147c87372d29" />

<img width="244" alt="image" src="https://github.com/user-attachments/assets/9714baa0-b234-49cb-a15f-4a5d02f87b07" />

**Comment by tgreyuk** [comment]
Thanks @DominicGBauer for feedback :)

The issue here is that the strikethough has been created  with the Unicode `\u0336` (combining long stroke overlay). What is happening is that the default font doesn't support this character so is falling back to a font that does.

Was probably a bit hasty getting this out and  I think a plan B is required here. 

New proposed solution is to add a class name to the relevant sidebar item. This could then be handled properly by css. Like so:

`typedoc-sidebar.cjs` (autogenerated sidebar)

<img width="493" alt="Screenshot 2025-01-07 at 17 39 31" src="https://github.com/user-attachments/assets/8d091424-8577-4943-b58a-8d98f4d3f1d5" />

Generated html:

<img width="478" alt="Screenshot 2025-01-07 at 17 48 41" src="https://github.com/user-attachments/assets/ba81d80b-cc67-46b7-80a7-e4dfd69d7c00" />

<br />

`docusaurus.config.js` (added by user)

<img width="428" alt="Screenshot 2025-01-07 at 17 42 41" src="https://github.com/user-attachments/assets/1a6c8385-2ebe-420e-a5af-e2b091c250ac" />

`css/typedoc.css`  (added by user - could also be added to existing custom css if exists )

<img width="360" alt="Screenshot 2025-01-07 at 17 42 57" src="https://github.com/user-attachments/assets/8a519bff-d95f-401c-a20b-9dbe85da3e1f" />

output:

<img width="237" alt="Screenshot 2025-01-07 at 17 56 16" src="https://github.com/user-attachments/assets/0f9d0efc-6932-4edf-b5d0-8b85cfc72a3d" />

Does this seem reasonable? It gives users greater flexibility - could make the class name configurable if useful.

**Comment by DominicGBauer** [comment]
Hi @tgreyuk yes please that would be brilliant! Thanks a lot :)

**Comment by tgreyuk** [comment]
@DominicGBauer `docusaurus-plugin-typedoc@1.2.1`. See https://www.typedoc-plugin-markdown.org/plugins/docusaurus/sidebar#deprecated-sidebar-items.

**Comment by DominicGBauer** [comment]
This is perfect :D! Thank you very much @tgreyuk!
<img width="582" alt="image" src="https://github.com/user-attachments/assets/df64fb8a-980f-4072-aaa5-65772979cb78" />
